cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConditionConfigurer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.workflow;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.schedule.SchedulableProgramType;
+import co.cask.cdap.api.workflow.WorkflowAction;
+import co.cask.cdap.api.workflow.WorkflowConditionConfigurer;
+import co.cask.cdap.api.workflow.WorkflowConditionNode;
+import co.cask.cdap.api.workflow.WorkflowNode;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Default implementation of the {@link WorkflowConditionConfigurer}.
+ * @param <T> the type of the parent configurer
+ */
+public class DefaultWorkflowConditionConfigurer<T extends WorkflowConditionAdder>
+  implements WorkflowConditionConfigurer<T>, WorkflowConditionAdder {
+
+  private final T parentConfigurer;
+  private final List<WorkflowNode> ifBranch = Lists.newArrayList();
+  private final List<WorkflowNode> elseBranch = Lists.newArrayList();
+  private List<WorkflowNode> currentBranch;
+  private boolean addingToIfBranch = true;
+  private final Predicate<Map<String, String>> predicate;
+
+  public DefaultWorkflowConditionConfigurer(T parentConfigurer, Predicate<Map<String, String>> predicate) {
```

Comment:
You only need the predicate class name in the configurer, no need to carry the instance around.